test.py

```diff
@@ -0,0 +1,125 @@
+ENV_DIR = FILE_DIR.joinpath("kubernetes_env")
+COMPILER_BINARY = COMPILER_DIR.joinpath("target/debug/dtc")
+QUERY_DIR = COMPILER_DIR.joinpath("example_queries")
+UDF_DIR = COMPILER_DIR.joinpath("example_udfs")
+
+KUBE_QUERIES = [
+    pytest.param("count.cql", ["count"]),
+    pytest.param("breadth_histogram.cql", ["histogram"]),
+    pytest.param("height_histogram.cql", ["histogram"]),
+    pytest.param("response_code_count.cql", ["count"]),
+    pytest.param("response_size_avg.cql", ["avg"]),
+    pytest.param("return.cql", []),
+    pytest.param("return_height.cql", []),
+]
+# test names
+KUBE_IDS = [i.values[0] for i in KUBE_QUERIES]
+
+SIM_QUERIES = [
+    pytest.param("count.cql", ["count"]),
+    pytest.param("breadth_histogram.cql", [
+                 "histogram"], marks=pytest.mark.xfail),
+    pytest.param("height_histogram.cql", [
+                 "histogram"], marks=pytest.mark.xfail),
+    pytest.param("response_code_count.cql", ["count"]),
+    pytest.param("response_size_avg.cql", ["avg"], marks=pytest.mark.xfail),
+    pytest.param("return.cql", [], marks=pytest.mark.xfail),
+    pytest.param("return_height.cql", [], marks=pytest.mark.xfail),
+]
+# test names
+SIM_IDS = [i.values[0] for i in SIM_QUERIES]
+
+
+class TestClassKubernetes:
+    @classmethod
+    def setup_class(cls):
+        """ setup any state specific to the execution of the given class (which
+        usually contains tests).
+        """
+        result = kube_env.setup_bookinfo_deployment("MK", False)
+        assert result == util.EXIT_SUCCESS
+        # start the modified bookinfo app, ignore the result for now
+        result = kube_env.install_modded_bookinfo()
+
+    @pytest.mark.parametrize("query_file,query_udfs", KUBE_QUERIES, ids=KUBE_IDS)
+    def test_deployment(self, query_file, query_udfs):
+        # generate the filter code
+        cmd = f"{COMPILER_BINARY} "
+        cmd += f"-q {QUERY_DIR.joinpath(query_file)} "
+        for query_udf in query_udfs:
+            udf_file = UDF_DIR.joinpath(query_udf)
+            udf_file = udf_file.with_suffix(".cc")
+            cmd += f"-u {udf_file} "
+        result = util.exec_process(cmd)
+        assert result == util.EXIT_SUCCESS
+        # build the filter
+        result = kube_env.build_filter(kube_env.FILTER_DIR)
+        assert result == util.EXIT_SUCCESS
+        # deploy the filter
+        result = kube_env.refresh_filter(kube_env.FILTER_DIR)
+        assert result == util.EXIT_SUCCESS
+
+    @classmethod
+    def teardown_class(cls):
+        """ teardown any state that was previously setup with a call to
+        setup_class.
+        """
+        kube_env.stop_kubernetes("MK")
+
+
+class TestClassSimulator:
+    filter_dir = COMPILER_DIR.joinpath("rust_filter")
+    target_filter_dir = SIM_DIR.joinpath("libs/rust_filter")
+
+    @pytest.mark.parametrize("query_file,query_udfs", SIM_QUERIES, ids=SIM_IDS)
+    def test_deployment(self, query_file, query_udfs):
```

Comment:
I think this is going to fail because it doesn't copy the types file over.  I know that stuff is not yet implemented so maybe we can just copy it automatically into the sim/src folder for now, and mark it down and find a more permanent solution later.

Reply:
hmm I'd rather have the test fail than copy the types_file in. Fixing that is the next todo for me after this pull request works. So far only two filter pass, which might be because of the types.

Reply:
Yeah, that's just because the right type file is in.  I'd mark them as test fail until we have a working solution, because it's really just coincidence that they're working.  That sounds like a good plan.